ouroboros-consensus/src/Ouroboros/Consensus/ChainSyncServer.hs

```diff
@@ -23,6 +25,8 @@ import qualified Ouroboros.Storage.ChainDB.API as ChainDB
 import           Ouroboros.Consensus.Block
 import           Ouroboros.Consensus.Util.ResourceRegistry (ResourceRegistry)
 
+-- | Used to advertise the tip of the server
+type Tip b = (Point b, BlockNo)
```

Comment:
Let's move this to a new `Consensus.ChainSync` module, and have both the server and the client be monomorphic in that tip.

Reply:
If we're going to do that, let's maybe introduce `Consensus.ChainSync.Server`, `Consensus.ChainSync.Client`, and `Consensus.ChainSync.???` (and similarly for blockfetch?).

Reply:
And maybe turn `Tip` into a proper ADT.

Reply:
`encodeTip` and `decodeTip` should move to the new module.